server/headless-agent/src/puppeteer/puppetMaster.ts

```diff
@@ -93,6 +93,8 @@ export class PuppetMaster extends EventEmitter {
         // Overriding this to use a setTimeout based version.
         await this.page.evaluate(() => {
             (window as any).setImmediate = (callback: any) => {
+                // TODO: possible bug?
+                // eslint-disable-next-line @typescript-eslint/no-implied-eval
```

Comment:
This was a workaround to fix a problem with puppeteer library. Without this, our loader will not work since all the setImmediate() we use will be intercepted by puppeteer.